internal/ocm/resource/cluster.go

```diff
@@ -108,6 +108,42 @@ func (c *Cluster) CreateNodes(autoScalingEnabled bool, replicas *int64, minRepli
 	return nil
 }
 
+func (c *Cluster) CreateHcpNodes(replicas *int64, computeMachineType *string,
+	labels map[string]string, availabilityZones []string, multiAZ bool, workerDiskSize *int64) error {
+	nodes := cmv1.NewClusterNodes()
+	nodes.Compute(int(*replicas))
+	if computeMachineType != nil {
+		nodes.ComputeMachineType(
+			cmv1.NewMachineType().ID(*computeMachineType),
+		)
+	}
+
+	if workerDiskSize != nil {
+		nodes.ComputeRootVolume(
+			cmv1.NewRootVolume().AWS(
+				cmv1.NewAWSVolume().Size(int(*workerDiskSize)),
+			),
+		)
+	}
+
+	if labels != nil {
+		nodes.ComputeLabels(labels)
+	}
+
+	if availabilityZones != nil {
+		if err := validations.ValidateAvailabilityZonesCount(multiAZ, len(availabilityZones)); err != nil {
```

Comment:
multiAZ is not supported for HCP Node pools

Reply:
HCP nodes creation is always single zone

Reply:
This is fixed